### PR TITLE
Qt: draw border in OnroadWindow

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by pd0wm** [comment]
Can we make sure the border is non transparent? Currently it still relies on black bars around the camera view to render correctly. However, with the map open it looks wrong.

**Comment by pd0wm** [comment]
Can we extend the camera view all the way to the edge? That way there are no black borders in split view.

2021-07-14-123051_1395x1083_scrot

**Comment by deanlee** [comment]
Is this OK? I am traveling and there is no device to test now.

**Comment by pd0wm** [comment]
Now the video is completely gone. The scissor is probably necessary.

**Comment by pd0wm** [comment]
Also saw the video was now gone on master, so that must be a problem with my machine. So disregard.

**Comment by pd0wm** [comment]
Can we also continue the lane lines behind the border? With the split the lane lines are cut off too soon.

**Comment by deanlee** [comment]
how about now?

**Comment by pd0wm** [comment]
Yes, looks good! Feel free to remove draft status.

@adeebshihadeh does this look good to you?

**Comment by adeebshihadeh** [comment]
I don't think we want this as part of the onroad alerts. The border should be part of the layout (a margin in the base layout everything else is drawn on), otherwise you have to think about what's hidden by the border.

**Comment by deanlee** [comment]
> I don't think we want this as part of the onroad alerts. The border should be part of the layout (a margin in the base layout everything else is drawn on), otherwise you have to think about what's hidden by the border.

moved to OnroadWindow,looks better now.

File: selfdrive/ui/qt/onroad.cc [hunk 1]
```diff
@@ -7,42 +7,49 @@
...
+  nvg->updateState(s);
+  QColor bgColor = alerts->updateState(s);
+  if (bg != bgColor) {
+    bg = bgColor;
+    update();
+  }
+}
 
 void OnroadWindow::offroadTransition(bool offroad) {
+  alerts->offroadTransition(offroad);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.cc:52]: this should use signals/slots

File: selfdrive/ui/qt/onroad.cc [hunk 2]
```diff
@@ -54,17 +61,24 @@ void OnroadWindow::offroadTransition(bool offroad) {
...
-      split->addWidget(m);
-
-      map = m;
+      map = new MapWindow(settings);
+      split->addWidget(map);
     }
-
+  }
+  if (map) {
+    ((MapWindow*)map)->offroadTransition(offroad);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.cc:69]: same as above

File: selfdrive/ui/qt/onroad.h [hunk 3]
```diff
@@ -17,19 +17,19 @@ class OnroadAlerts : public QWidget {
   Q_OBJECT
 
 public:
-  OnroadAlerts(QWidget *parent = 0) {};
+  OnroadAlerts(QWidget *parent = 0) : QWidget(parent) {};
+  QColor updateState(const UIState &s);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.h:21]: updateState should be a standard slot all UI widgets have, don't like that this one returns a QColor. we can add a getter for the color or just abstract out the logic to get the background color from the alert status

Comment (deanlee) [reply selfdrive/ui/qt/onroad.h:21]: Yes,that's better

**Comment by deanlee** [comment]
It feels a bit strange to use getter

**Comment by deanlee** [comment]
@adeebshihadeh  what do you think?

**Comment by adeebshihadeh** [comment]
The border now frequently gets out of sync with the alerts.

**Comment by deanlee** [comment]
sorry! my bad. forgot to call update after this change

**Comment by adeebshihadeh** [comment]
Looks good now. Can we keep the background until we get a frame though? I think the black frame on startup looks worse now.

**Pull request merged**